File: Assignment_8/Assignment_8/controllers/FriendListViewController.m


#import "FriendListViewController.h"

@interface FriendListViewController ()
{
    NSString *CellIdentifier;
    NSArray *fetchedObjects;
    
    __weak IBOutlet UITableView *tableViewFriendsList;
}
@end

@implementation FriendListViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    //[self fetchFromCoreData];
    CellIdentifier = @"cell";
    [tableViewFriendsList registerNib:[UINib nibWithNibName:@"CustomizedTableViewCell" bundle:nil] forCellReuseIdentifier:CellIdentifier];
    
   // [tableViewFriendsList reloadData];
}
-(void)viewWillAppear:(BOOL)animated{
     [self fetchFromCoreData];
    [tableViewFriendsList reloadData];
}



- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)buttonAddDetails:(id)sender {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    AddDetailsViewController*addDetailsViewController=  [storyboard instantiateViewControllerWithIdentifier:@"AddDetailsViewController"];
    addDetailsViewController.isFriends = YES;
    [self.navigationController pushViewController:addDetailsViewController animated:true];
}


- (NSManagedObjectContext *)managedObjectContext {
    NSManagedObjectContext *context = nil;
    id delegate = [[UIApplication sharedApplication] delegate];
    if ([delegate performSelector:@selector(managedObjectContext)]) {
        context = [delegate managedObjectContext];
    }
    return context;
}


-(void)fetchFromCoreData {
    
    
    NSManagedObjectContext *context = [self managedObjectContext];
    
    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
    
    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Friend" inManagedObjectContext:context];
    
    [fetchRequest setEntity:entity];
    
    NSError *error = nil;
    fetchedObjects = [[NSArray alloc]init];
    fetchedObjects = [context executeFetchRequest:fetchRequest error:&error];
}


#pragma mark -TableViewDataSource


- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    
    return [fetchedObjects count];
    
    
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    return 61;
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    CustomizedTableViewCell * cell;
    
    if (cell == nil) {
        cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    }
    Friend *friend  =[fetchedObjects objectAtIndex:indexPath.row];
    cell.labelContactOrLocation.text = [NSString stringWithFormat:@"%@",friend.contactNo];
    cell.labelNameOrOrganisation.text = friend.name;
    
    
    return cell;
}


@end

File: Assignment11/Assignment11/controllers/FriendListViewController.swift


import Foundation

import UIKit
let cellIdentifier = "cell"
var arrayOfPersons : NSArray!
var person = Person()

class FriendListViewController: UIViewController ,UITableViewDataSource , UITableViewDelegate  {
    
    @IBOutlet weak var tableViewFriendList: UITableView!

    //Mark: View Controller function
    
    override func viewDidLoad() {
        super.viewDidLoad()
       self.navigationController?.navigationBarHidden = false

    }
    
    override func viewWillAppear(animated: Bool) {
        
   arrayOfPersons = fetchAllRecordsFromEntity("Person")
    
    tableViewFriendList.reloadData()
        
    }
    
    

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
       
    }
    
   //MARK :- Table view
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int
    {
       return(arrayOfPersons.count)
       
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell{
              
        var cell:UITableViewCell! =
        tableView.dequeueReusableCellWithIdentifier(cellIdentifier) as? UITableViewCell
               if (cell == nil)
        {
            person.firstName = arrayOfPersons.objectAtIndex(indexPath.row) .valueForKey("firstName") as! String
            person.contactNo = arrayOfPersons.objectAtIndex(indexPath.row) .valueForKey("contactNo") as! String

            println(person.firstName)
            cell = UITableViewCell(style: UITableViewCellStyle.Subtitle,
                reuseIdentifier: cellIdentifier)
             cell.textLabel?.text = person.firstName
             cell.detailTextLabel?.text = person.contactNo
        }
        
        return cell;
        
        
    }
    
    //MARK :- button click function
    
    @IBAction func buttonAddDetailsOnPress(sender: AnyObject) {
        let addDetailsViewController
        = self.storyboard!.instantiateViewControllerWithIdentifier("AddDetailsViewController")
            as! AddDetailsViewController
        addDetailsViewController.friendsOrOrganisation = true
        self.navigationController!.pushViewController(addDetailsViewController, animated: true)
        
    }

}

